dual.go

```diff
@@ -229,9 +230,10 @@ func dvBind(op Op, inputs []*dualValue) (retVal *dualValue, err error) {
 
 	var ret Value
 	if ret, err = op.Do(vals...); err == nil {
-		retVal = dvUnit(ret)
+		return dvUnit(ret), nil
+	} else {
+		return nil, errors.Wrap(err, opDoFail)
```

Comment:
perhaps wrap the `Op` name into the errors as well? (all `Op`s implement  `fmt.Stringer`)